### PR TITLE
#160046190 Fix Request Page Reloading

### PR DESCRIPTION
**Pull request by albert85**
#### What does this PR do?
Fix request page reloading bug

#### Description of Task to be completed?
- Change "my requests" and "my approvals" links to Navlink
- Fix logout dropdown
- Create drawer navigation using CSS
- Fix request modal details overlay
- Fix request dropdown toggling when "my requests" link is clicked
- Fix the hamburger menu icon disappearing when "open" button is clicked

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into the project directory
- Run the server `make start`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#160046190](https://www.pivotaltracker.com/story/show/160046190)

#### Screenshots (if appropriate)
screen shot 2018-09-03 at 10 32 24 am

#### Questions:
N/A

### CODE REVIEW

File: src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js [hunk 1]
```diff
@@ -101,4 +101,15 @@ describe('<ApprovalsPage>', () => {
     wrapper.unmount();
   });
 
+  it('should handle handleOverlay method', ()=> {
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js]: @albert85 , I can see that in your test for the navbar, you are calling the instance method directly which does correctly represent the way the method is  interacted with. I believe clicking the logout button is what calls the method. I suggest you change this and simulate the button click instead to check if the method was actually called.

File: src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js [hunk 2]
```diff
@@ -101,4 +101,15 @@ describe('<ApprovalsPage>', () => {
...
+    const wrapper = shallow(<Approvals {...props} />)
+    wrapper.instance().handleOverlay();
+    expect(wrapper.state('hideOverlay')).toBe(false);
+  });
+  it('should handle handleShowDrawer method', ()=> {
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js]: The same feedback applies here

File: src/views/ApprovalsPage/index.jsx [hunk 3]
```diff
@@ -34,7 +36,7 @@ class Approvals extends Base {
   renderApprovalPage (hideClass2,leftPaddingClass, requests, pagination, hideClass, hideClass3, selectedLink ) {
     return (
       <div className="mdl-layout__content full-height">
-        <div className="mdl-grid mdl-grid--no-spacing full-height">
+        <div className="mdl-grid mdl-grid--no-spacing full-height" onClick={this.handleHideLogoutDropdown} id="logout" role="presentation">
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/index.jsx]: Please can you move the other props aside the className to a new line?

File: src/views/ApprovalsPage/index.jsx [hunk 4]
```diff
@@ -54,15 +56,17 @@ class Approvals extends Base {
...
-    const { hideNotificationPane, hideSideBar, openSearch, selectedLink } = this.state;
+    const { hideNotificationPane, hideSideBar, openSearch, selectedLink, hideOverlay } = this.state;
     let [hideClass, leftPaddingClass] = hideNotificationPane? ['hide', '']: ['', 'pd-left'];
     const hideClass2 = hideSideBar ? 'hide mdl-cell--hide-desktop' : '';
     const hideClass3 = hideSideBar ? '' : 'hide mdl-cell--hide-desktop';
     const { requests, pagination } = requestsData;
+    const overlayClass = hideOverlay ? 'block': 'none';
     return(
       <div>
+        <div className="side_overlay" style={{display: `${overlayClass}`}} role="button" onClick={this.handleOverlay} onKeyPress={() => {}} tabIndex="0" />
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/index.jsx]: Same feedback as the last applies here

File: src/views/RequestsPage/RequestsPage.scss [hunk 5]
```diff
@@ -1,60 +1,74 @@
...
+        margin-top: 14px;
+        font-family: 'DIN Pro';
+        color: #4f4f4f;
+    }
+    &__divider {
+        margin-top: 41px;
+        margin-bottom: -57px;
+        border-top: 1px solid lightgrey;
+    }
+}
```
Comment (amarachukwu-agbo) [inline src/views/RequestsPage/RequestsPage.scss]: Please insert a new line at the end of this file

File: src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js (hunk 1 above)
Comment (albert85) [reply src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js]: @amarachukwu-agbo there is a test that was previously written for logout functionality, that why i did unit testing of the instance

File: src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js (hunk 2 above)
Comment (albert85) [reply src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js]: @amarachukwu-agbo same reason as above here

File: src/views/ApprovalsPage/index.jsx (hunk 4 above)
Comment (albert85) [reply src/views/ApprovalsPage/index.jsx]: @amarachukwu-agbo done

File: src/views/RequestsPage/RequestsPage.scss (hunk 5 above)
Comment (albert85) [reply src/views/RequestsPage/RequestsPage.scss]: @amarachukwu-agbo thanks for the review, feedback implemented

**Comment by samuelkubai** [comment]
@albert85 please resolve conflicts on this.

**Comment by babadee001** [comment]
LGTM @samuelkubai

**Pull request merged**